server/src/main/java/org/opensearch/action/search/TransportCreatePITAction.java

```diff
@@ -0,0 +1,341 @@
+import org.opensearch.common.Strings;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.io.stream.NamedWriteableRegistry;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.index.shard.ShardId;
+import org.opensearch.search.SearchPhaseResult;
+import org.opensearch.search.SearchService;
+import org.opensearch.search.SearchShardTarget;
+import org.opensearch.search.internal.ShardSearchContextId;
+import org.opensearch.tasks.Task;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.*;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+/**
+ * Transport action for creating PIT reader context
+ */
+public class TransportCreatePITAction extends HandledTransportAction<CreatePITRequest, SearchResponse> {
+
+    public static final String CREATE_PIT = "create_pit";
+    private final TimeValue CREATE_PIT_TEMPORARY_KEEP_ALIVE = new TimeValue(30, TimeUnit.SECONDS);
+    private final SearchService searchService;
+    private final TransportService transportService;
+    private final SearchTransportService searchTransportService;
+    private final ClusterService clusterService;
+    private final TransportSearchAction transportSearchAction;
+    private final NamedWriteableRegistry namedWriteableRegistry;
+
+    @Inject
+    public TransportCreatePITAction(
+        SearchService searchService,
+        TransportService transportService,
+        ActionFilters actionFilters,
+        SearchTransportService searchTransportService,
+        ClusterService clusterService,
+        TransportSearchAction transportSearchAction,
+        NamedWriteableRegistry namedWriteableRegistry
+    ) {
+        super(CreatePITAction.NAME, transportService, actionFilters, in -> new CreatePITRequest(in));
+        this.searchService = searchService;
+        this.transportService = transportService;
+        this.searchTransportService = searchTransportService;
+        this.clusterService = clusterService;
+        this.transportSearchAction = transportSearchAction;
+        this.namedWriteableRegistry = namedWriteableRegistry;
+    }
+
+    @Override
+    protected void doExecute(Task task, CreatePITRequest request, ActionListener<SearchResponse> listener) {
+        SearchRequest searchRequest = new SearchRequest(request.getIndices());
+        searchRequest.preference(request.getPreference());
+        searchRequest.routing(request.getRouting());
+        searchRequest.indicesOptions(request.getIndicesOptions());
+        searchRequest.allowPartialSearchResults(request.isAllowPartialPitCreation());
+
+        final StepListener<SearchResponse> createPitListener = new StepListener<>();
```

Comment:
this shouldn't be SearchResponse. We are listening for pitId. It just comes wrapped up in the SearchResponse object.

Reply:
+1